docs/initializing.rst

```diff
@@ -555,6 +556,42 @@ point into the system. It will handle the setup and initialization of the config
 and start up a Toolkit session once all the required pieces have been initialized and set up.
 
 
+Distributing application code from your site
+============================================
+
+If you want to distribute custom application code that may only be accessible for
+users on your local network or from developers who have access keys to the Git
+repositories, you can upload individual bundles to Shotgun and then take over the ``populate_bundle_cache_entry`` and ``can_cache_bundle`` methods of the
+``core/bootstrap.py`` hook.
+
+Here's a suggested setup:
+
+.. image:: ./resources/initializing/bundle_custom_entity.png
+
+As you can see, a non project custom entity has been configured to store Toolkit bundles. The
+most important column here is Descriptor, which is the field the bootstrap hook will use
+to determine if a bundle is available in Shotgun or not.
+
+Once the bundles have been uploaded, you can implement the ``core/bootstrap.py`` hook.
+
+.. literalinclude:: ../tests/fixtures/descriptor_tests/with_create_descriptor_hook/core/hooks/bootstrap.py
+   :language: python
+   :start-after: #documentationStart
+   :end-before: #documentationEnd
+
+Once you are done, you can zip your configuration and its custom bootstrap and upload it to a
+``PipelineConfiguration`` entity in Shotgun. Now everytime a user bootstraps into this pipeline
+configuration, they will download the configuration and cache the Toolkit core and all application
+bundles through the hook.
```

Comment:
can we add a paragraph here which talks about the build process? Say something like - if you are using this to distribute your own apps to remote users, you can just do it manually. If your users are disconnected from the Internet and you want to use this as a mechanism to distribute everything, you can write a build script which analyzes the dependencies of a toolkit configuration and uploads each one to Shotgun.

Reply:
i agree with shouldn't oversell it or provide extensive receipes. I guess what i am trying to say is that we should just _mention_ that you can automate this process easily and for any complex workflow, this is what we would advise people to do. Just so they understand the thinking and know it is a possibility. Does that make sense?

Reply:
But people can't *easily* automate this. They have no APIs to enumerate the bundles in a config. They either need to access the pipeline configuration object which is not public or they need to use the yaml module and parse their environments files recursively in search of mappings that look like descriptors.